pytorch_lightning/trainer/training_loop.py

```diff
@@ -863,6 +864,12 @@ def _accumulated_batches_reached(self):
     def _num_training_batches_reached(self):
         return (self.trainer.batch_idx + 1) == self.trainer.num_training_batches
 
+    def should_accumulate(self):
```

Comment:
can this be a property?

Ideally `_accumulated_batches_reached()` and `_num_training_batches_reached()` would also be properties, but that's off the scope of this PR

Reply:
Yes, you are entirely right 👍 I will do this in the next PR.